### PR TITLE
Add docusaurus docs site and content

### PR DESCRIPTION
**Pull request by mattfrances**
## Description
Creates a Docusaurus website for the Restyle package's documentation - containing all content that is currently in the readme. 

Next steps:
1. Replace readme with a simple description of the package and usage example
2. Deploy to GitHub Pages

## Reviewers’ hat-rack :tophat:
Test that the docs site runs locally
- `cd docusaurus`
- `yarn`
- `yarn start`
- Go to [localhost](http://localhost:3000/) and go through the pages

## Checklist

- [ ] I have added a changelog entry following the [Keep a Changelog](http://keepachangelog.com/en/1.0.0/) guidelines.

### CODE REVIEW

**Comment by mattfrances** [comment]
> How will this page be deployed? I don't see any workflows related to the docs. It also would have been _nice_ to get a preview of the docusaurus page but not sure how difficult that is to do in the open source setting, so we can leave it as a possible improvement.

The docs site will be deployed using GitHub Pages. I'll be separating the addition of the related workflow to a separate PR as I currently don't have the access rights to edit the repository's settings and I think it should be in a separate unit of work. The preview of the docusaurus page would be great, I agree with leaving it as a possible improvement!

File: documentation/dev.yml
Comment (mattfrances) [reply documentation/dev.yml]: I've removed it in the latest commit, although I noticed an issue when doing so. The [docusaurus-shopify-theme](https://github.com/Shopify/docusaurus-shopify-theme/blob/main/package.json#L16) requires a node version of 16.14.0 to be properly installed - without node v16.14, the `yarn` command doesn't work. I've reached out to the maintainers and a node version update should be coming soon. For the meantime, I've added some guidance on running the docs locally in the [Readme](https://github.com/Shopify/restyle/pull/215/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5).

**Review by fortmarek — COMMENTED** [review]
> I'll be separating the addition of the related workflow to a separate PR as I currently don't have the access rights to edit the repository's settings and I think it should be in a separate unit of work.

I don't necessarily dislike too much to merge this before that is ready, although the merge won't be of much value _until_ we can deploy the site. Apart from that, we should also add a CI check like [this](https://github.com/Shopify/flash-list/blob/main/.github/workflows/website.yml) one that will check the docs actually build before being pushed to `main`.

**Comment by mattfrances** [comment]
> I don't necessarily dislike too much to merge this before that is ready, although the merge won't be of much value _until_ we can deploy the site.

Good point, in that case I'll get a PR up with the workflow to deploy the docs site to github pages, and will merge this PR only once that one is ready. That way we can still have separate PRs for each unit of work, and can ensure that the docs site provides value from the start.

**Review by fortmarek — APPROVED** [review]
🚀 

As a follow-up to this work (and once we deploy the docs), we should update `README.md` to point to the new microsite instead (and remove most of the content that currently resides in the readme)

**Pull request merged**
